addons/mail/static/src/models/thread/thread.js

```diff
@@ -1291,7 +1295,7 @@ function factory(dependencies) {
          * @returns {mail.message[]}
          */
         _computeNeedactionMessages() {
-            return [['replace', this.messages.filter(message => message.isNeedaction)]];
+            return [['replace', this.messagesAsOriginThread.filter(message => message.isNeedaction)]];
```

Comment:
This slightly changes the meaning of this field, so that current name "needaction messages" is misleading.
Isn't it best to introduce another field `needactionMessagesAsOriginThread` and remove the `needactionMessages` in master?

Reply:
`markNeedactionMessagesAsRead` becomes `markNeedactionMessagesAsOriginThreadAsRead`?

Reply:
It's making too much duplicate code actually. `markNeedactionMessagesAsRead`, `_computeLastNeedactionMessage`, `needactionMessages` itself, diff on every line using it...

Since we won't ever need the distinction between the two, I think the current PR is best for 14. We can just rename in master if we want, but we probably should not duplicate everything here.

Reply:
> diff on every line using it...

Which is good IMO, because we know exactly which part of code is affected by this slight change and rightfully. If somewhere we feel it's actual all needaction messages and not just as origin thread, we can do the change. Right now we should trust our gust, and reading code is misleading because we need to make extra effort to translate "needaction messages" to "needaction messages as origin thread". Very likely to make mistake whenever we make changes in stable that closely relate to these needaction messages...

> Since we won't ever need the distinction between the two, I think the current PR is best for 14. We can just rename in master if we want, but we probably should not duplicate everything here.

Never ever assume we don't need the distinction in the future. Also if a variable is poorly named, it's good reason to rename (or make another field with proper name, in this case).

This is a very poor move keeping bad name just to reduce diff. Only case I would accept this is if we won't ever touch the code again in v14, guaranteed not being the case... So definitely have name clarification here.

Reply:
Done, feel free to double check because I was confident in the previous diff but I could have easily missed one rename here.